backend/src/middleware/logger.middleware.ts

```diff
@@ -0,0 +1,34 @@
+import { UserService } from '@/user/user.service';
+import { Injectable, NestMiddleware } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { NextFunction, Request, Response } from 'express';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  constructor(
+    private jwt: JwtService,
+    private userService: UserService
+  ) {}
+  async use(req: Request, res: Response, next: NextFunction) {
+    const cookie = req.cookies['jwt'];
+    if (!cookie) {
+      return res.status(401).json({ message: 'Unauthorized User' });
+    }
+    try {
+      const { id: user_id } = await this.jwt.verifyAsync(cookie);
+      await this.userService.findOne({ where: { user_id } });
+
+      if (
+        req.headers['cookie'] &&
+        req.headers['cookie'].includes(`jwt=${cookie}`)
+      ) {
+        return next();
+      }
+
+      return res.status(401).json({ message: 'Unauthorized User' });
+    } catch (error) {
+      console.error(error);
+      return res.status(401).json({ message: 'Invalid Token' });
+    }
+  }
```

Comment:
- i think there is no need of these:

https://github.com/framgia/sph-els-mark/blob/f796b10db694bfdc9c06e4b8af42dc17688bb4da/backend/src/answer/answer.controller.ts#L37

https://github.com/framgia/sph-els-mark/blob/f796b10db694bfdc9c06e4b8af42dc17688bb4da/backend/src/answer/answer.controller.ts#L55-L62

- kindly remove this
https://github.com/framgia/sph-els-mark/blob/f796b10db694bfdc9c06e4b8af42dc17688bb4da/backend/src/answer/answer.controller.ts#L63

(Kindly apply changes to other files which has these)

Reply:
there are many files using the following code. 
pls apply the same thing you did to [this](https://github.com/framgia/sph-els-mark/pull/34#discussion_r1449012156)

```javascript
if (user_id) { 
   return await this.answerService.save({ 
     attempts: attempts.attempt_id, 
     words: wordId.word_id, 
     answer: body.answer, 
     question_no: body.question_no, 
   }); 
 } 
```